drivers/pwm/pwm_nrfx.c

```diff
@@ -214,23 +197,28 @@ static int pwm_nrfx_pin_set(struct device *dev, u32_t pwm,
 		} else {
 			nrf_gpio_pin_set(pwm);
 		}
-	}
 
-	/* Check if all channels are off (duty 0% or 100%) */
-	if (!any_channel_active(data)) {
-		nrfx_pwm_stop(&config->pwm, false);
+		/* Check if any other channel is active. If not, stop PWM. */
```

Comment:
imo, something that was previously here was better in terms of readability. I mean, encapsulating algorithm for checking if anything is active into a function.

Something like:
```
if (!any_channel_active(data)) {
   nrfx_pwm_stop(&config->pwm, false);
}
```

Reply:
Agreed. I'll restore `any_channel_active`.

Reply:
Extracted this part of code into `any_other_channel_is_active()`.